Img protocol (#62)
vtex-apps/store-image: docs/README.md

All CSS Handles available for the Image block are the ones available for the `slider-layout` block. Take a look at the Customization section in the [**Slider Layout documentation**](https://vtex.io/docs/app/vtex.slider-layout).
Note that the `image-slider` uses our `vtex.slider-layout` app, so all the CSS namespaces defined by it are also available for `image-slider`. Take a look at [Slider-Layout](https://vtex.io/docs/app/vtex.slider-layout).

> ## Image Protocol Implementation

Reviewer: ```suggestion
## Implementing Image Protocol 
```

> 

Reviewer: ```suggestion
The Image Protocol is an app that displays personalized images to use alongside the Store Image app in your store's images. Follow the steps below to implement the Image Protocol in your code.
```

> In the `manifest.json` we added Session Client app and also the Image Protocol app in the dependency list:

Reviewer: ```suggestion
1. In your theme's `manifest.json`, add the Session Client app and also the Image Protocol app as a dependency:
```

```json
 "dependencies": {
    "vtex.session-client": "1.x",
    "vtex.image-protocol": "0.x"
  }
```

> The interface `ImageTypes.ts` is modified to have the isMobile and imageProtocolId props:

Reviewer: @LeslieJRP, I didn't understand this part. Could you help me by answering the following questions?
1. Should the developer add the `ImageTypes.ts` on their code?
2. Is this an example of how should it look? 
3. Is this a step-by-step that the user should follow to modify the `ImageTypes.ts`?
Thanks!

Author: Hi @Mariana-Caetano 
1. The developer doesn't need to add anything in the ImageTypes.ts, this is already done
2. The code should be like this, no need to modify anything
3. No, the user doesn't need to modify anything in the code, to modify the props the user can use the json file or the site editor

Hope this helps, let me know if you need any other clarification.
Thanks!

```ts
export interface ImageSchema {
  isMobile?: boolean
  imageProtocolId? : string
  ...
}
```

> To be able to personalize a specific image the props defined in the interface are also exposed in the Image component: 

Reviewer: ```suggestion
2. To personalize a specific image, the props defined in the interface must be exposed in the Image component as the example below: 
```

```tsx
const{
  isMobile=false,
  imageProtocolId='',
  ...
}=props
``` 

> >**NOTE**: The prop imageProtocolId can be modified from the Site Editor.

Reviewer: ```suggestion
> ⚠️
> 
> The prop `imageProtocolId` can be modified via Site Editor.
```

Reviewer: How can we modify the prop via Site Editor?

Author: If you go to the Site Editor and check the List of images, selecting any of the images you will see the field 'Image Protocol Id'
<img width="1359" alt="Captura de pantalla 2022-06-10 a las 12 10 56" src="https://user-images.githubusercontent.com/104071100/173044274-09025b46-9c50-4418-8671-ab064beca67f.png">

> To implement the URLs from the image protocol example we have created inside the `react/graphql` the `getImgUrl.gql` that will use the resolver in the `image-protocol` app.

Reviewer: ```suggestion
3. Implement the URLs from the image protocol by creating inside the `react/graphql` the `getImgUrl.gql` that will use the resolver in the `image-protocol` app. See the example below:
```

```graphql
query getImage($userId: String!, $imageProtocolId: String!) {
    getImage(userId: $userId, imageProtocolId: $imageProtocolId) @context(provider: "vtex.image-protocol"){
    url,
    urlMobile
  }
}
```

> In order to get the user Id we use the useRenderSession and the SessionSuccess from the vtex.session-client package in the `Image.tsx`. Also to use the query created inside the graphq folder we also import it as well as the useQuery.

Reviewer: ```suggestion
4. Get the user Id using the `useRenderSession` and the `SessionSuccess` from the `vtex.session-client `package in the `Image.tsx`. Also, to use the query created inside the GraphQL folder, import the query as well as the `useQuery`.
```

```tsx
import { useQuery } from 'react-apollo'
import GET_ImgUrl from './graphql/getImgUrl.gql'

import { SessionSuccess, useRenderSession } from 'vtex.session-client'
```

> Now as the Image component has the imageProtocolId prop if it has an Id and we can get the user Id we can use the query defined in `getImgUrl.gql` that receives these two variables and is skipped if no user id is provided.

Reviewer: ```suggestion
> ⚠️
> 
> As the Image component has the `imageProtocolId` prop if it has an Id, you can get the user Id and use the query defined in `getImgUrl.gql` that receives these two variables. If no user id is provided, skip this suggestion.
```

Reviewer: Could you explain more about this part?

> We set the src for this imgElement depending on the response of the query and also if this is for mobile or not. Other image elements that don't have the image protocol Id or the result of the query has the URLs as null we then set the default URLs.

Reviewer: These two sentences could be a step-by-step

Author: It's not a step by step because it's done in the code, I explained here how it works
